### PR TITLE
DOCSP-42523 Proxy Configuration Support

### PR DESCRIPTION
**Pull request by ajhuh-mdb**
## DESCRIPTION
- Documents proxy configuration environment variables

## STAGING
https://preview-mongodbajhuhmdb.gatsbyjs.io/mongodb-shell/DOCSP-42523-proxy-config-support/connect/#proxy-settings

## JIRA
https://jira.mongodb.org/browse/DOCSP-42523

## BUILD LOG
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=66bfb7988e811ae67289d17d

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/includes/table-proxy-env-variables.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+       ``example.com:8080`` with TLS enabled.
+
+       .. code-block:: sh 
+          :copyable: false 
+          
+          export MONGODB_PROXY=https://example.com:8080
+
+   * - ``HTTP_PROXY``
+
+     - Proxy connections to ``http://`` URLs. HTTP connections primarily relate
```
Comment (jmd-mongo) [inline source/includes/table-proxy-env-variables.rst:25]: [suggestion]

Consider simplifying language a bit here. Spitballing: "HTTP connections are mostly used for OIDC authentication." or simliar.

File: source/includes/table-proxy-env-variables.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+       ``example.com:8080``:
+
+       .. code-block:: sh 
+          :copyable: false 
+
+          export HTTP_PROXY=http://example.com:8080 
+
+   * - ``HTTPS_PROXY``
+
+     - Proxy connections to ``https://`` URLs. HTTPS connections primarily
```
Comment (jmd-mongo) [inline source/includes/table-proxy-env-variables.rst:42]: Similar comment as on line 25.

**Review by jmd-mongo — COMMENTED** [review]
Thanks for these improvements, @ajhuh-mdb! I've left some suggestions and a question for your consideration.

Thanks,
Joe

File: source/includes/table-proxy-env-variables.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+       proxying.
+
+     - The following example sets the ``NO_PROXY`` environment variable to 
+       bypass the proxy for connections to ``localhost`` and 
+       ``internal-db.example.com``.
+
+       .. code-block:: sh 
+          :copyable: false 
+
+          export NO_PROXY=localhost, internal-db.example.com
```
[the remark this answers is not in the document]
Comment (ajhuh-mdb) [reply source/includes/table-proxy-env-variables.rst:82]: Ah thanks for the catch! Tested and updated w/o the space.

File: source/includes/table-proxy-env-variables.rst (hunk 1 above)
Comment (ajhuh-mdb) [reply source/includes/table-proxy-env-variables.rst:25]: Updated, thanks for the suggestion!

**Comment by ajhuh-mdb** [comment]
Thanks for the first pass @jmd-mongo! This should be ready for another look.

**Review by jmd-mongo — APPROVED** [review]
Thanks for these changes, @ajhuh-mdb! This one LGTM.

Thanks,
Joe

**Comment by ajhuh-mdb** [comment]
Hi @addaleax 👋 Whenever you get the chance, could you review this docs PR for the proxy configuration env variables on mongosh? TIA!

File: source/connect.txt [hunk 4]
```diff
@@ -357,6 +357,24 @@ following command:
...
+.. include:: /includes/table-proxy-env-variables.rst
+
+.. note:: 
+
+   ``mongosh`` supports the following proxy types: 
+   
+   - Socks5 proxies
+   - HTTP proxies 
+   - CONNECT proxies
+   - :abbr:`PAC (proxy auto-configx)` URLs that resolve to one of the previously
```
Comment (addaleax) [inline source/connect.txt:375]: typo: 
```suggestion
   - :abbr:`PAC (proxy auto-config)` URLs that resolve to one of the previously 
```

**Review by addaleax — APPROVED** [review]
Awesome, this looks exactly right!

File: source/connect.txt (hunk 4 above)
Comment (ajhuh-mdb) [reply source/connect.txt:375]: Ack thanks for the catch! Fixed.

**Pull request merged**